pyrdp/mitm/FastPathMITM.py

```diff
@@ -37,6 +37,34 @@ def onClientPDUReceived(self, pdu: FastPathPDU):
         if self.state.forwardInput:
             self.server.sendPDU(pdu)
 
+        if not self.state.loggedIn:
+            for event in pdu.events:
+                if isinstance(event, FastPathScanCodeEvent):
+                    self.onScanCode(event.scanCode, event.isReleased, event.rawHeaderByte & 2 != 0)
+
     def onServerPDUReceived(self, pdu: FastPathPDU):
         if self.state.forwardOutput:
-            self.client.sendPDU(pdu)
\ No newline at end of file
+            self.client.sendPDU(pdu)
+
+    def onScanCode(self, scanCode: int, isReleased: bool, isExtended: bool):
```

Comment:
We should probably store some of the special characters and output them in the logs/json. For example: backspace and tab.

A user has more chances of understanding `P<\b>Passw0rd!` rather than `PPassw0rd!`. I'm not too opinionated about what format it should be but since we are already seeing `\x00` in some fields, we could just pass it like that, so backspace would be `\x08` and tab `\x09`. That's just a suggestion, something fancier would be ok too.

Reply:
Here is an example candidate currently. I tokenized each action the same way to simplify logging / parsing accuracy. 
`MonPass<ctrl-a><\b>Passw0rd!`